src/opentracing/Tracer.ts

```diff
@@ -155,13 +155,24 @@ class ThundraTracer extends Tracer {
         const tags: any = {};
         let span;
         const rootTraceId = fields.rootTraceId ? fields.rootTraceId : Utils.generateId();
-
         const parentContext = fields.parentContext ? fields.parentContext : Utils.getParentContext(fields.references);
+        const parentSpan = fields.childOf ? fields.childOf : this.getActiveSpan();
 
         if (!fields.propagated && (parentContext && !this.activeSpans.get(parentContext.spanId))) {
             throw new Error('Invalid spanId : ' + parentContext.spanId);
         }
 
+        const stopReporting: boolean = fields.stopReporting;
+        // Skip reporting if
+        // - it is explicitly specified
+        // - or recording of its parent is skipped
+        // - or recording is stopped from its parent
+        const skipReporting: boolean =
+            (fields && fields.skipReporting)
```

Comment:
null check does not necessary for 'fields' in this line, it can be removed or it can be placed before first usage

Reply:
removed that code part as it was not necessary for our new solution